### ISSUE TITLE
value of Maybe is shown as `null`, when in fact it isn't

### ISSUE THREAD

**Issue by freder** [origamitower/folktale#191]
new `folktale` user here.
↓ this has caused some confusion for me while playing with the library:

### Steps to reproduce

```js
console.log(Maybe.of(Infinity));
console.log(Maybe.of(Infinity).unsafeGet());
console.log(Maybe.of(NaN));
console.log(Maybe.of(NaN).unsafeGet());
```

#### Expected behaviour

should print

```
folktale:Maybe.Just({ value: Infinity })
Infinity
folktale:Maybe.Just({ value: NaN })
NaN
```

#### Observed behaviour

prints

```
folktale:Maybe.Just({ value: null })
Infinity
folktale:Maybe.Just({ value: null })
NaN
```

**Comment by robotlolita** [comment]
I'm going to try releasing a new version with this fix this weekend.

**Comment by robotlolita** [comment]
This has been released as `folktale@2.3.0`.
